jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriver.java

```diff
@@ -85,8 +87,20 @@ Connection createConnection(String url, Properties info) throws SQLException {
 
         if (url.startsWith(DATAPROC_JDBC_HIVE_URL_SCHEMA)) {
             try {
-                String hiveURL = formatHiveUrl(url);
-            } catch (InvalidURLException e) {
+                HiveJdbcConnectionOptions params = parseHiveUrl(url);
+                String myEndpoint = String.format("%s-dataproc.googleapis.com:443", params.region());
+
+                // Configure the settings for the cluster controller client.
+                ClusterControllerSettings clusterControllerSettings =
+                        ClusterControllerSettings.newBuilder().setEndpoint(myEndpoint).build();
+
+                ClusterControllerClient clusterControllerClient =
+                        ClusterControllerClient.create(clusterControllerSettings);
+
+                DataprocInfo clusterInfo = new DataprocInfo(params, clusterControllerClient);
+
+                String hiveURL = clusterInfo.toHiveJdbcUrl();
+            } catch (IOException e) {
                 throw new SQLException(e.getCause());
```

Comment:
Pass original exception in the `new SQLException`, otherwise you may lose valuable information in its stacktrace.